components/content-service/pkg/service/workspace-service.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package service
+
+import (
+	"context"
+	"strings"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/gitpod-io/gitpod/common-go/tracing"
+	"github.com/gitpod-io/gitpod/content-service/api"
+	"github.com/gitpod-io/gitpod/content-service/pkg/storage"
+	"github.com/opentracing/opentracing-go"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// WorkspaceService implements WorkspaceServiceServer
+type WorkspaceService struct {
+	cfg storage.Config
+	s   storage.PresignedAccess
+}
+
+// NewWorkspaceService create a new content service
+func NewWorkspaceService(cfg storage.Config) (res *WorkspaceService, err error) {
+	s, err := storage.NewPresignedAccess(&cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &WorkspaceService{cfg, s}, nil
+}
+
+// DownloadUrlWorkspace provides a URL from where the content of a workspace can be downloaded from
+func (cs *WorkspaceService) DownloadUrlWorkspace(ctx context.Context, req *api.DownloadUrlWorkspaceRequest) (resp *api.DownloadUrlWorkspaceResponse, err error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "DownloadUrlWorkspace")
+	span.SetTag("user", req.OwnerId)
+	span.SetTag("workspaceId", req.WorkspaceId)
+	defer tracing.FinishSpan(span, &err)
+
+	blobName := cs.s.BackupObject(req.WorkspaceId, storage.DefaultBackup)
+
+	info, err := cs.s.SignDownload(ctx, cs.s.Bucket(req.OwnerId), blobName, &storage.SignedURLOptions{
+		// ContentType: "application/tar+gzip",
+		ContentType: "*/*",
```

Comment:
AFAIK GCP is very particular when it comes to content types, especially the one that the client wich accesses the pre-signed URL sends. Especially now that we no longer proxy-pass the download, but redirect we've introduced a dependency to the client here.

- Do we have to fix a content type, what happens if we don't?
- Does workspace download work in Chrome/Firefox/Safari?

Reply:
The problem is that workspace download does not work at all at the moment (see #3152). That's why I cannot test this. Probably we need to fix #3152 first before we could get this into `main`.

Reply:
Current implementation works with Chrome and Firefox. Cannot test with Safari (no Mac available).